compiler/src/dmd/dmacro.d

```diff
@@ -424,3 +428,30 @@ Largstart:
     //printf("extractArg%d('%.*s') = '%.*s'\n", n, cast(int)end, p, cast(int)marg.length, marg.ptr);
     return v;
 }
+
+/*****************************************
+ * Determine length of UTF string.
+ * Params:
+ *      p = point to start of string
+ * Returns: number of bytes in UTF character. Returns 1 on invalid UTF start.
+ */
+@safe
+int utfStride(scope const(char)* p) @nogc nothrow pure
```

Comment:
Instead of duplicating this function, a common function should be imported from `dmd.root.utf`. Otherwise, this function needs a unittest and correct documentation. ("length of an UTF string" is wrong, it returns the number of UTF-8 code units in a code point based on the first one)

Reply:
I fixed the documentation and added some unittests. Generally, you're right that one shouldn't repeat oneself. In this case, I wanted dmacros.d to be usable as a macro text processor on its own, and utfStride() is a trivial function. root.utf is a substantial module. In this case, it's justifiable for a tiny bit of redundancy.